gtsam/nonlinear/Values-inl.h

```diff
@@ -292,6 +292,11 @@ namespace gtsam {
    struct handle_matrix<Eigen::Matrix<double, M, N>, true> {
      Eigen::Matrix<double, M, N> operator()(Key j, const Value* const pointer) {
        try {
+        // TODO(fan): Investigate how to make dynamic_cast work for Eigen on Mac with dylibs
+        // std::cout << "Wanted: " << typeid(const GenericValue<Eigen::Matrix<double, M, N>>&).name()
+        //  << ", " << typeid(const GenericValue<Eigen::Matrix<double, M, N>>&).hash_code() << std::endl;
+        //  std::cout << "Actual: " << typeid(*pointer).name() << ", " << typeid(*pointer).hash_code() << std::endl;
+         
```

Comment:
We need to solve this issue but in a different PR. Please fix README so that on Mac: "On mac, you have to statically build gtsam to use the wrapper." etc